rust/rbac-registration/src/cardano/cip509/utils/extract_key.rs

```diff
@@ -0,0 +1,83 @@
+//! Functions for extracting public key from X509 and C509 certificates with additional
+//! verification.
+
+use anyhow::{anyhow, Context};
+use c509_certificate::c509::C509;
+use ed25519_dalek::{VerifyingKey, PUBLIC_KEY_LENGTH};
+use oid_registry::{Oid, OID_SIG_ED25519};
+use x509_cert::Certificate as X509Certificate;
+
+/// Returns `VerifyingKey` from the given X509 certificate.
+///
+/// # Errors
+///
+/// Returns an error if the signature algorithm is not supported and
+/// the public key cannot be extracted.
+pub fn x509_key(cert: &X509Certificate) -> anyhow::Result<VerifyingKey> {
```

Comment:
I think these two functions `x509_key`,  `c509_key`,  could be in the same file as `CertOrPk`

Reply:
I think it is better to have these 2 in its own utility.
It is more like a helper function to extract a public key of subject from certificate.
In case anyone have a cert and want to get those pk, can just call this function